### DISCUSSION TITLE
useBalance not working , after update

### DISCUSSION THREAD

**Discussion by ghost** [wevm/wagmi#1396] [Q&A] [answered]
as i have said useBalance is not working , 

useBalance for ETH
` const coinBalance = useBalance({
    addressOrName: address,
    chainId: chain?.id,
    watch: Boolean(chain?.id),
    enabled: isConnected,
  });`

useBalance for ERC20
`  const ercBalance = useBalance({
    addressOrName: address,
    token: tokenAddress,
    chainId: chain?.id,
    watch: Boolean(chain?.id),
    enabled: isConnected,
  });`

this was not returning anything 
this was the response
`{
    "error": null,
    "fetchStatus": "idle",
    "isError": false,
    "isFetched": false,
    "isFetchedAfterMount": false,
    "isFetching": false,
    "isIdle": true,
    "isLoading": false,
    "isRefetching": false,
    "isSuccess": false,
    "status": "idle",
    "internal": {
        "dataUpdatedAt": 0,
        "errorUpdatedAt": 0,
        "failureCount": 0,
        "isFetchedAfterMount": false,
        "isLoadingError": false,
        "isPaused": false,
        "isPlaceholderData": false,
        "isPreviousData": false,
        "isRefetchError": false,
        "isStale": true
    }
}`

but if i use useContractRead the response is correct

the code
` const data = useContractRead({
    abi: erc20ABI,
    address: spentAddress,
    functionName: "balanceOf",
    args: [address],
    enabled: isConnected,
    watch: true,
  });`

the response of useContractRead
`{
    "data": {
        "type": "BigNumber",
        "hex": "0xcd6e12674c0101e48388"
    },
    "error": null,
    "fetchStatus": "idle",
    "isError": false,
    "isFetched": true,
    "isFetchedAfterMount": true,
    "isFetching": false,
    "isIdle": false,
    "isLoading": false,
    "isRefetching": false,
    "isSuccess": true,
    "status": "success",
    "internal": {
        "dataUpdatedAt": 1669798905615,
        "errorUpdatedAt": 0,
        "failureCount": 0,
        "isFetchedAfterMount": true,
        "isLoadingError": false,
        "isPaused": false,
        "isPlaceholderData": false,
        "isPreviousData": false,
        "isRefetchError": false,
        "isStale": true
    }
}`

have i wrote something wrong ? , any help is appricated because i can get ERC-20 balances from contract call but not ETH balance 
i need ETH balance in my daap

**Comment by tmm** [comment, marked as answer]
My guess is you need to use `address` instead of `addressOrName` for `useBalance`. This [changed in 0.8](https://wagmi.sh/react/migration-guide#usebalance).

**Reply by ghost** [reply]
thanks ,
